### ISSUE TITLE
Sparse Vectors write in a non-useful way.

### ISSUE THREAD

**Issue by thesuperzapper** [tensorflow/ecosystem#42]
When using the spark-tensorflow-connector, and writing a dataframe containing a sparse vector to the tfrecords format, we write the sparse representation, rather than the more useful dense representation.
`(3,[1,2],[1.0,1.0])` rather than `[0.0, 1.0, 1.0]`

I am not sure what the default behaviour should be, or how we can handle this more elegantly, because just blindly densifying sparse vectors could cause problems. 

However, tensorflow doesn't like sparse data types, so we should at least give the option of densifying. 

**NOTE:** A current workaround is to define a UDF which densifies the vector. 

**Here is an example: (Written in PySpark)**
```
##Define a toy corpus.
df = spark.createDataFrame([
    (0, "Spark is cool"),
    (1, "Tensorflow is cool also"),
    (2, "Why doesn't Tensorflow support normal data formats"),
    (3, "Tensorflow eats my GPU's")
], ["id", "sentence"])


##Tokenize the corpus.
from pyspark.ml.feature import Tokenizer
tokenizer = Tokenizer(inputCol="sentence", outputCol="words")
df = tokenizer.transform(df)

##Fit a CountVectorizerModel from the corpus.
from pyspark.ml.feature import CountVectorizer
cv = CountVectorizer(inputCol="words", outputCol="features", minDF=2.0)
df = cv.fit(df).transform(df)

##Display Dataset
df.printSchema()
df.show(truncate=False)

#Write as TFRecords.
df.repartition(1).write.format("tensorflow").save("/storage/example/example.tfrecords")
```

**Here is the output of the above code:**
```
root
 |-- id: long (nullable = true)
 |-- sentence: string (nullable = true)
 |-- words: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- features: vector (nullable = true)
+---+--------------------------------------------------+----------------------------------------------------------+-------------------------+
|id |sentence                                          |words                                                     |features                 |
+---+--------------------------------------------------+----------------------------------------------------------+-------------------------+
|0  |Spark is cool                                     |[spark, is, cool]                                         |(3,[1,2],[1.0,1.0])      |
|1  |Tensorflow is cool also                           |[tensorflow, is, cool, also]                              |(3,[0,1,2],[1.0,1.0,1.0])|
|2  |Why doesn't Tensorflow support normal data formats|[why, doesn't, tensorflow, support, normal, data, formats]|(3,[0],[1.0])            |
|3  |Tensorflow eats my GPU's                          |[tensorflow, eats, my, gpu's]                             |(3,[0],[1.0])            |
+---+--------------------------------------------------+----------------------------------------------------------+-------------------------+
```

**Here is the output of** `cat /storage/example/example.tfrecords/*`:

```
P9�
}

id


sentence


Spark is cool
*
words!

WrappedArray(spark, is, cool)
#
features

(3,[1,2],[1.0,1.0])��>�����x
�

id

B
sentence6
4
2Why doesn't Tensorflow support normal data formats
S
wordsJ
H
FWrappedArray(why, doesn't, tensorflow, support, normal, data, formats)

features


(3,[0],[1.0])��=��5V
�

id

'
sentence

Tensorflow is cool also
5
words,
*
(WrappedArray(tensorflow, is, cool, also)
)
features

(3,[0,1,2],[1.0,1.0,1.0])j�����}C
�

id

(
sentence

Tensorflow eats my GPU's
6
words-
+
)WrappedArray(tensorflow, eats, my, gpu's)

features


(3,[0],[1.0])�XA�
```

**Comment by skavulya** [comment]
@thesuperzapper Sorry for the delay in getting back to you. Thanks for bringing up this issue. We will work on a fix.

**Comment by skavulya** [comment]
@thesuperzapper We added support for Vectors in https://github.com/tensorflow/ecosystem/pull/53. Please let us know if this resolved your issue once the PR is merged.

Scala code:
```scala
import org.apache.spark.sql.types._
import org.apache.spark.sql.Row
import org.apache.spark.ml.feature.{CountVectorizer, CountVectorizerModel}
import org.apache.spark.ml.feature.{RegexTokenizer, Tokenizer}

val rdd = sc.parallelize(Seq(Row(0, "Spark is cool"),
    Row(1, "Tensorflow is cool also"),
    Row(2, "Why doesn't Tensorflow support normal data formats"),
    Row(3, "Tensorflow eats my GPU's")))
val schema = StructType(Seq(StructField("id", IntegerType), StructField("sentence", StringType)))
val df = spark.createDataFrame(rdd, schema)


val tokenizer = new Tokenizer().setInputCol("sentence").setOutputCol("words")
val df2 = tokenizer.transform(df)

//Fit a CountVectorizerModel from the corpus.

val cv = new CountVectorizer().setInputCol("words").setOutputCol("features").setMinDF(2.0)
val df3 = cv.fit(df2).transform(df2)
df3.write.format("tfrecords").option("recordType", "Example").save("corpus.tfrecord")

val importedDf = spark.read.format("tfrecords").option("recordType", "Example").load("corpus.tfrecord")
importedDf.show()
```
+--------------------+---------------+--------------------+---+
|               words|       features|            sentence| id|
+--------------------+---------------+--------------------+---+
|   [spark, is, cool]|[0.0, 1.0, 1.0]|       Spark is cool|  0|
|[tensorflow, is, ...|[1.0, 1.0, 1.0]|Tensorflow is coo...|  1|
|[why, doesn't, te...|[1.0, 0.0, 0.0]|Why doesn't Tenso...|  2|
|[tensorflow, eats...|[1.0, 0.0, 0.0]|Tensorflow eats m...|  3|
+--------------------+---------------+--------------------+---+

**Comment by skavulya** [comment]
@jhseu This issue is resolved #53  by and can be closed. Thanks!
